azurerm_private_dns_a_record_resource - currently validate the `name` property by allowing `@`s

stephybun opened the pull request:
Fixes #12406

tombuildsstuff commented on helpers/validate/strings.go:
```diff
@@ -43,5 +43,9 @@ func LowerCasedString(i interface{}, k string) ([]string, []error) {
 		return nil, []error{fmt.Errorf("%q cannot contain whitespace", k)}
 	}
 
+	if strings.ContainsAny(v, "@") {
```
I initially went [through the Azure naming rules since _some_ services allow these iirc](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules) thinking this was used more widely than it is - but since this functions only used for Private DNS, it'd probably be worth moving this into the Private DNS package at the same time, since that's the only usage at the moment:

![Screenshot 2021-08-19 at 09 21 41](https://user-images.githubusercontent.com/666005/130025579-09436329-3471-4f4e-af4a-0c431644a89f.png)

stephybun replied:
Makes sense, thanks for the input, i've moved this validation to the Private DNS package.

tombuildsstuff reviewed (changes requested):
one thought about refactoring this but this otherwise LGTM 👍

katbyte reviewed (approved):
Thanks @stephybun - LGTM now 🏗️

calloes commented:
Why is this changed ? apex dns records are valid in private dns zones. Our existing code now fails.

Internal DNS zone can have a valid root record represented by @

[31m
[1m[31mError: [0m[0m[1m"name" cannot contain @[0m

[0m  on .terraform/modules/private_dns/main.tf line 7, in resource "azurerm_private_dns_a_record" "zone_root_a_record":
   7:   name                = [4m"@"[0m
[0m
[0m[0m

stephybun commented:
Hello @calloes, yep you are right. We've reverted the validation on name that's causing your code to fail and added validation on the number of allowed records in this PR #13093 which will be released in v2.74.0 on Thursday.
